jetty-client/src/main/java/org/eclipse/jetty/client/http/HttpReceiverOverHTTP.java

```diff
@@ -38,7 +39,7 @@
 
 public class HttpReceiverOverHTTP extends HttpReceiver implements HttpParser.ResponseHandler
 {
-    private final HttpParser parser = new HttpParser(this);
+    private final HttpParser parser = new HttpParser(this, -1, HttpCompliance.RFC2616);
```

Comment:
why are you forcing the compliance mode here?  There is a lookup in the default constructor

Reply:
Hm yes, but we for instead use jetty as a server and a client in one application. Changing the parser mode to none strict per default is not really an option for us. Since it changes the behavior both for the client and the server at once.
And as the comments of the compliance mode state out, that RFC2616 is compatible down to http 0.9 where whitespaces weren't explicitly disallowed. Where LEGACY also make header parsing case sensitive, which is also not an option for us. 
In general, I think a client should be more weak and acceptable against none compliant HTTP servers. For instance, chrome accepts the response from the provided page of the issue and parses the HTTP header correctly.

Reply:
I think we want to leave the default compliance mode to RFC7230, which is the current spec and obsoletes RFC2616.   However there should be an easy way to programatically set the compliance mode on the client... @sbordet ?

Reply:
I can try to implement that the compliance mode is configurable on the jetty client itself and delegate the mode down to the parser creation if you like.

Reply:
Let's wait for guidance from @sbordet about how to do that, but it would be good to include that in this PR.